src/Fmt_ast.ml

```diff
@@ -2654,6 +2669,45 @@ and fmt_class_type_field c ctx (cf : class_type_field) =
     $ fmt_atrs )
 
 and fmt_cases c ctx cs =
+  let rec pattern_len ?(parens = 0) pat =
+    match pat.ppat_desc with
+    | Ppat_any -> Some 1
+    | Ppat_var {txt= s; _}
+     |Ppat_constant (Pconst_integer (s, _))
+     |Ppat_constant (Pconst_float (s, _))
+     |Ppat_construct ({txt= Lident s; _}, None) ->
+        Some (String.length s)
+    | Ppat_construct ({txt= Lident s; _}, Some arg) -> (
+      match pattern_len ~parens:2 arg with
+      | Some arg -> Some (parens + String.length s + 1 + arg)
+      | None -> None )
+    | Ppat_constant (Pconst_char _) -> Some 4 (* when the char is escaped *)
+    | Ppat_variant (s, None) -> Some (String.length s + 1)
+    | Ppat_variant (s, Some arg) -> (
+      match pattern_len ~parens:2 arg with
+      | Some arg -> Some (parens + 1 + String.length s + 1 + arg)
+      | None -> None )
+    | Ppat_constant (Pconst_string (s, _)) -> Some (String.length s + 2)
```

Comment:
Also, the second argument of `Pconst_string` is the delimiter of `{delim| ... |delim}` strings.